base/operators.jl

```diff
@@ -341,6 +341,16 @@ false
 julia> a === a
 true
 ```
+
+# Extended help
+It might be possible to construct a program that is actually able to distinguish
+two egal values. For example, one could heap-allocate two integers,
+then compare the value of a pointer pointing to their heap address.
+However, all such programs rely on observing behaviour outside Julia's allowed
+semantics, and so happen to work due to implementation details and/or triggering
```

Comment:
Reading this my first thought was: What are "Julia's allowed semantics", and where can I find out more.

I.e.: To me this doesn't clarify much. Mainly I am left with this message: "You can't distinguish egal values by the rules, which I am not telling you, just so you know: if you found a way it is cheating and you should not rely on it in any way."

But is that really helpful? Honest open question, I don't know. Since you opened the PR, presumably you feel this text (or at least parts of it) would have been helpful for you to know when you run into this?

Reply:
That's a fair summary. On the other hand, the current message is "you can't distinguish egal values", which is just false - you absolutely can, e.g. by comparing pointers to strings, as you did in the example above.
When I first read that docstring I tried the same thing as was confused - I then had to ask on Slack what was up with that. Since then, I've seen twice that people have asked the same question - those have been lost to the slack hole, but searching for the docstring of `===` on Discourse brings [this post up, with the same question](https://discourse.julialang.org/t/why-a-b-returns-true-when-a-and-b-are-single-elements-numbers-or-strings/69510?u=jakobnissen).
I.e. as I see it, the confusing thing is that the claim "you can't distinguish these two values with any program" appears superficially simple, but the statement's truth relies on complex language semantics which almost no user knows about.

So, I think either that statement should be removed from the docs, or else it should be properly explained what it means.
I would prefer explaining it, and possibly linking to the devdocs on UB which explains it in more detail.